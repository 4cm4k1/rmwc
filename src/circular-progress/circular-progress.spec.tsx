import React from 'react';
import { render } from '@testing-library/react';
import { CircularProgress } from './';

describe('CircularProgress', () => {
  it('renders', () => {
    const { asFragment } = render(<CircularProgress />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('can be sizes', () => {
    render(<CircularProgress size="xsmall" />);
    render(<CircularProgress size="small" />);
    render(<CircularProgress size="medium" />);
    render(<CircularProgress size="large" />);
    render(<CircularProgress size="xlarge" />);
    render(<CircularProgress size={72} />);
  });

  it('can be determinate', () => {
    render(<CircularProgress progress={0.3} />);
  });

  it('can have a different max / min', () => {
    render(<CircularProgress min={0} max={100} progress={30} />);
    render(<CircularProgress progress={-1} />);
    render(<CircularProgress progress={2} />);
  });
});
