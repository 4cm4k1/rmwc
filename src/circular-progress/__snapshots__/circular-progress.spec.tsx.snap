// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircularProgress renders 1`] = `
<DocumentFragment>
  <div
    aria-valuemax="1"
    aria-valuemin="0"
    class="rmwc-circular-progress rmwc-circular-progress--indeterminate"
  >
    <svg
      class="rmwc-circular-progress__circle"
      viewBox="0 0 24 24"
    >
      <circle
        class="rmwc-circular-progress__path"
        cx="12"
        cy="12"
        r="10"
      />
    </svg>
  </div>
</DocumentFragment>
`;
